phoenix-core/src/main/java/org/apache/phoenix/iterate/TableSnapshotResultIterator.java

```diff
@@ -88,19 +90,34 @@ public TableSnapshotResultIterator(Configuration configuration, Scan scan, ScanM
   }
 
   private void init() throws IOException {
-    RestoreSnapshotHelper.RestoreMetaChanges meta =
-        RestoreSnapshotHelper.copySnapshotForScanner(this.configuration, this.fs,
-            this.rootDir, this.restoreDir, this.snapshotName);
-    List<RegionInfo> restoredRegions = meta.getRegionsToAdd();
-    this.htd = meta.getTableDescriptor();
-    this.regions = new ArrayList<RegionInfo>(restoredRegions.size());
-
-    for (RegionInfo restoredRegion : restoredRegions) {
-      if (isValidRegion(restoredRegion)) {
-        this.regions.add(restoredRegion);
+    if (PhoenixConfigurationUtil.getMRSnapshotManagedInternally(configuration)) {
+      RestoreSnapshotHelper.RestoreMetaChanges meta =
+          RestoreSnapshotHelper.copySnapshotForScanner(this.configuration, this.fs, this.rootDir,
+                      this.restoreDir, this.snapshotName);
+      List<RegionInfo> restoredRegions = meta.getRegionsToAdd();
+      this.htd = meta.getTableDescriptor();
+      this.regions = new ArrayList<>(restoredRegions.size());
+      for (RegionInfo restoredRegion : restoredRegions) {
+        if (isValidRegion(restoredRegion)) {
+          this.regions.add(restoredRegion);
+        }
+      }
+    } else {
+      Path snapshotDir = SnapshotDescriptionUtils.getCompletedSnapshotDir(snapshotName, rootDir);
```

Comment:
Is the idea that the creator of this MR job restores this snapshot? If yes, I think that'd be inconvenient. How about we do it in the setup and let the InputFormat handle it? All user has to do is pass the snapshot name. Is there some issue with this approach?

Reply:
@bharathv creation was not an issue, I can plug it in easily, the problem was the cleanup part, which I was not able to figure out right spot to do that.

Reply:
@bharathv - I agree with your approach as the correct way forward, but since we're very close to a release think that work can be deferred to a future JIRA. The ability to turn off any automated snapshot handling is an easy fix to the immediate problem, and still useful even after we fix the automation problems later (for example if two jobs are part of a larger process and want to share a restored snapshot.) What do you think? 

@sakshamgangwar - if we do go with the current approach, we should probably change the name of the JIRA and commit to something more descriptive of what it actually does.

Reply:
Ok, I'm fine with this approach as long as the root problem is eventually fixed (perhaps raise a child task?). +1 on changing the commit message, I got confused by it myself.